File: src/main/java/Interfaceinstances.java
import java.util.HashSet;
import java.util.Set;

public class Interfaceinstances {

    public static class Widget {

        public interface Listener {
            void onClick(Widget w);
        }

        public final Set<Listener> listeners = new HashSet<>();

        public void addListener(Listener a) {
            listeners.add(a);
        }

        public void removeListener(Listener a) {
            listeners.remove(a);
        }

        public void simulateClick() {
            for(Listener l : listeners) {
                l.onClick(this);
            }
        }

    }
}

File: src/main/kotlin/InterfaceInstances.kt
import Interfaceinstances.Widget


fun main () {

    val myListener = { w: Widget -> println("Listener Called!")}

    val w = Widget()


    println("\nbefore adding any")
    println(w.listeners.size)

    w.addListener(myListener)

    println("\nafter adding one")
    println(w.listeners.size)

    w.removeListener(myListener)

    println("\nafter removing")
    println(w.listeners.size)
}